modules/candidate_parameters/jsx/DiagnosisEvolution.js

```diff
@@ -0,0 +1,223 @@
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data: data, formData: data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  /**
+   * Set form data
+   * @param {string} formElement
+   * @param {*} value
+   */
+  setFormData(formElement, value) {
+    let formData = this.state.formData;
+    formData[formElement] = value;
+    this.setState({
+      formData: formData,
+    });
+  }
+
+  /**
+   * @return {array}
+   */
+  formattedDiagnosisEvolution() {
+    const dxEvolution = this.state.data.diagnosisEvolution;
+    let formattedDxEvolution = [];
+    dxEvolution.map((record) => {
+      let formattedDiagnosis = [];
+      Object.entries(JSON.parse(record.Diagnosis)).map((entry) => {
+        const [fieldName, dx] = entry;
+        formattedDiagnosis.push(<p>{fieldName}: <strong>{dx}</strong></p>);
+      });
+      let sourceFields = [];
+      record.sourceField.split(',').map((field) => {
+        sourceFields.push(<p>{field}</p>);
+      });
+      const confirmed = record.Confirmed === 'Y' ?
+        <p style={{color: 'green', fontSize: '3rem', textAlign: 'center'}}>
+          &#10004;
+        </p> :
+        <p style={{color: 'red', fontSize: '3rem', textAlign: 'center'}}>
+          &#10007;
+        </p>;
+      formattedDxEvolution.push(
+        [
+          record.TrajectoryName,
+          record.Project,
+          record.OrderNumber,
+          record.visitLabel,
+          record.instrumentName,
+          sourceFields,
+          formattedDiagnosis,
+          confirmed,
+          record.LastUpdate,
+        ]
+      );
+    });
+    return formattedDxEvolution;
+  }
+
+  /**
+   * Render latest diagnosis element
+   * @param {*} latestDiagnosis
+   * @return {JSX} - React markup for the component
+   */
+  renderLatestDiagnosis(latestDiagnosis) {
+    let element = [];
+
+    latestDiagnosis.map((entry) => {
+      const projectName = this.state.data.projects[entry.ProjectID];
+      let diagnosis = [];
+      Object.entries(JSON.parse(entry.Diagnosis)).map((entry) => {
+        const [fieldName, dx] = entry;
+        diagnosis.push(
+          <StaticElement
+            key={fieldName}
+            label={fieldName}
+            text={dx}
+          />
+        );
+      });
+
+      element.push(
+        <FieldsetElement
+          key={entry.DxEvolutionID}
+          legend={<h5>{projectName} - {entry.Name}</h5>}
+          class='col-md-6'
+        >
+          {diagnosis}
+        </FieldsetElement>
+      );
+    });
+    return element;
+  }
+
+  /**
+   * Renders the React component.
+   *
+   * @return {JSX} - React markup for the component
+   */
+  render() {
+    if (this.state.error) {
+        return <h3>An error occured while loading the page.</h3>;
+    }
+
+    if (!this.state.isLoaded) {
+        return <Loader/>;
+    }
+
+    const latestDiagnosis = this.state.data.latestProjectDiagnosis.length > 0 ?
```

Comment:
```suggestion

    const latestConfirmedProjectDiagnosisID = this.state.data.latestConfirmedProjectDiagnosis.length > 0 ?  this.state.data.latestConfirmedProjectDiagnosis[0].DxEvolutionID : null;
    const latestProjectDiagnosisID = this.state.data.latestProjectDiagnosis.length > 0 ?
    this.state.data.latestProjectDiagnosis[0].DxEvolutionID : null;
    
    const latestDiagnosis = (this.state.data.latestProjectDiagnosis.length > 0 &&
    latestConfirmedProjectDiagnosisID != latestProjectDiagnosisID) ?
```

Reply:
If I'm remembering correctly, the `latestConfirmedProjectDiagnosis` and `latestProjectDiagnosis` will have an entry for each of the candidate's projects. What happens in this case if there is more than one project (even if only 1 trajectory is defined for each project)? By explicitly accessing the first array position [0], you're only fixing the display for the first project. 

Not sure if my comment is clear.

Reply:
yeah you're clear! good point, i will think about this :)

Reply:
sent a PR to your branch fixing this! https://github.com/jesscall/Loris/pull/1